src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala

```diff
@@ -129,13 +128,12 @@ class DepositServlet(app: EasyDepositApiApp)
     (for {
       uuid <- getUUID
       path <- getPath
-      _ <- isMultipart
-      newFileItems <- getFileItems
+      _ <- getContentTypeIfMultipart
+      newFileItems <- fileMultiParams.values.flatten // TODO fileParams.values gives the first file of each form-field, why/when/which
```

Comment:
is `.valuesIterator.flatten` not lazy enough? In that case you don't need the `toStream` here.
Not sure though, haven't tested it...